gglasso/solver/single_admm_solver.py

```diff
@@ -7,154 +7,184 @@
 from scipy.sparse.csgraph import connected_components
 from scipy.linalg import block_diag
 
-
 from .ggl_helper import prox_od_1norm, phiplus, prox_rank_norm
 
 
-def ADMM_SGL(S, lambda1, Omega_0 , Theta_0 = np.array([]), X_0 = np.array([]), \
-             eps_admm = 1e-5 , rho= 1., max_iter = 1000, verbose = False, measure = False, latent = False, mu1 = None):
+def ADMM_stopping_criterion(Omega, Theta, L, X, S, lambda1, latent=False, mu1=None):
+    assert Omega.shape == Theta.shape == S.shape
+    assert S.shape[0] == S.shape[1]
+
+    if not latent:
+        assert np.all(L == 0)
+
+    (p, p) = S.shape
+
+    #new formula
+
+    eps_rel = 1e-3
+    eps_abs = 1e-3
+
+    e_pri = np.sqrt(p)*eps_abs + eps_rel*max(np.linalg.norm(Omega), np.linalg.norm(Theta), np.linalg.norm(X))
+    e_dual = np.sqrt(p)*eps_abs + eps_rel*np.linalg.norm(X)
+
+    term4 = 0
+    if latent:
```

Comment:
For the latent variable SGL problem, you can derive the stopping criteriona as well. The formulation is 
min_{Omega, Thtea, L} -log det (Omega) + <S,Omega> + lambda1 ||Theta||_1 + mu1 ||L||_\star
subject to Omega = Theta - L
and || . ||_\star is the nuclear norm